File: SOL_InevntoryManagement_Ver_5.0/InevntoryManagement/wwwroot/Amount.js

$("#PaymentOnCash").change(function () {
    let amountbill = amountclass()
    $("#Dues").val((amountbill.TotalAmount - amountbill.PaymentOnCash))



});
$("#Discount").change(function () {
    let amountbill = amountclass()
    $("#NetAmount").val((amountbill.GrossAmount - amountbill.Discount))


})

$("#Transport").change(function () {
    let amountbill = amountclass()
    let totalbill = 0

    totalbill = parseInt((parseInt(amountbill.TotalAmount) + parseInt(amountbill.Transport) + parseInt(amountbill.Others)))
    $("#GrossAmount").val(totalbill)
    $("#NetAmount").val(totalbill)
})

$("#Others").change(function () {
    let amountbill = amountclass()
    let totalbill = 0

    totalbill = parseInt((parseInt(amountbill.TotalAmount) + parseInt(amountbill.Transport) + parseInt(amountbill.Others)))
    $("#GrossAmount").val(totalbill)
    $("#NetAmount").val(totalbill)

})


function amountclass() {
    let _amount = {
        TotalAmount: $("#TotalAmount").val(),
        PaymentOnCash: $("#PaymentOnCash").val(),
        Dues: $("#Dues").val(),
        Transport: $("#Transport").val(),
        Others: $("#Others").val(),
        GrossAmount: $("#GrossAmount").val(),
        Discount: $("#Discount").val(),
        NetAmount: $("#NetAmount").val(),
        PaymentTypeId: $("#PaymentTypeId").val(),


    }

    return _amount
}


//calculate total product purchage price amount
function calculatePurchageProduct() {
    let tr = $("#productlist tr")
    $("#TotalAmount").val('0')
    if ($(tr).length > 0) {
        var amount = 0;
        $(tr).each(function (i, item) {
            amount = (amount + parseInt($(this).find("td:last").prev().text()));
        })

        $("#TotalAmount").val(amount) //product cost


        //Call amountclass To Get Update Total AMount After Adding New Product   
        let _amountclass = amountclass();

        let grossamount = (parseInt(_amountclass.TotalAmount) + parseInt(_amountclass.Transport) + parseInt(_amountclass.Others))
        let netamount = (grossamount - parseInt(_amountclass.Discount))
        $("#GrossAmount").val(grossamount)
        $("#NetAmount").val(netamount)
        $("#Dues").val(parseInt(_amountclass.TotalAmount) - parseInt(_amountclass.PaymentOnCash))
    }


}

File: SOL_IneventoryManagement/BussinessAccessLayer/Model/Amount.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace BussinessAccessLayer.Model
{
   public class Amount
    {
        [Key]
        public int Id { get; set; }
        public int TrID { get; set; }
        public int TrType { get; set; }
        public int TotalAmount { get; set; }
        public int PaymentOnCash { get; set; }

        public int Dues { get; set; }
        public int Transport { get; set; }
        public int Others { get; set; }
        public int GrossAmount { get; set; }
        public int Discount { get; set; }
        public int NetAmount { get; set; }
        public int PaymentTypeId { get; set; }

        public virtual List<MasterDetail> MasterDetails { get; set; } = new List<MasterDetail>();

    }
}
